src/actions/archiveThought.ts

```diff
@@ -54,13 +55,34 @@ export const pathAndRankToArchive = (
 
 /** Moves the thought to =archive. If the thought is already in =archive, permanently deletes it.
  *
- * @param path     Defaults to cursor.
+ * @param path                 Defaults to cursor.
+ * @param archivePathReference If true and the path points to a note with =path, also archives the target thought.
  */
-const archiveThought = (state: State, options: { path?: Path }): State => {
+const archiveThought = (state: State, options: { path?: Path; archivePathReference?: boolean }): State => {
```

Comment:
Adding the note logic to `archiveThought` has the same problem described above: You are making a low-level, generic function responsible for case-specific behavior. Previously, `archiveThought` did not need to know anything about notes, which was a good thing. 

Furthermore, `archivePathReference` is not orthogonal with `path`; there are preconditions, tightly coupling this to the caller in ways that are a bit difficult to infer. So it's not the best parameter design.